src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+                                {
+                                    // if it is too complex to remove (param validation, error handling, side effects, not everything assigned)
+                                    // we won't delete, and give a warning
+                                    // TODO: if parameter defaults in constructor, move those to primary constructor
+                                    // TODO: Change warning string
+                                    modifiedMembers.Add(
+                                        constructor.WithAdditionalAnnotations(WarningAnnotation.Create("Placeholder Constructor warning")));
+                                }
+                                continue;
+                            }
+
+                            if (constructorParamTypes.Length == 1 &&
+                                constructorParamTypes.First() == classDeclaration.Identifier.ToString())
+                            {
+                                // TODO: Check to see whether it's worth deleting
+                            }
+
+                            // TODO: Can potentially refactor statements which initialize properties with a simple expression
+                            // (not using local variables) to be moved into the this args
+                            var thisArgs = SyntaxFactory.ArgumentList(
+                                SyntaxFactory.SeparatedList(
+                                    Enumerable.Repeat(
+                                        SyntaxFactory.Argument(
+                                            SyntaxFactory.LiteralExpression(SyntaxKind.DefaultLiteralExpression)),
+                                        propertiesToMove.Length)));
+
+                            // TODO: look to see if there is already initializer (base or this)
+
+                            var modifiedConstructor = constructor
+                                .WithInitializer(SyntaxFactory.ConstructorInitializer(SyntaxKind.ThisConstructorInitializer, thisArgs));
+
+                            modifiedMembers.Add(modifiedConstructor);
+                            continue;
+                        }
+
+                    case OperatorDeclarationSyntax op:
+                        {
+                            // keep track of equality overloads so we can potentially remove them later
+                            var opKind = op.OperatorToken.Kind();
+                            if (opKind == SyntaxKind.EqualsEqualsToken)
+                            {
+                                equals = op;
+                            }
+                            else if (opKind == SyntaxKind.ExclamationEqualsToken)
+                            {
+                                notEquals = op;
+                            }
+
+                            break;
+                        }
```

Comment:
imo, this sort of optimization really isn't necessary.  just have a separate function call that does the work around the equality operators.  in other words, it's fine to look through the list of members multiple times.  nothing is really gained from trying to do everything in a single pass, and it def makes thsi entire function much more unweildy and diffifult to track.